src/library_dylink.js

```diff
@@ -359,7 +359,14 @@ var LibraryDylink = {
     loadedLibsByName: {},
     // handle  -> dso; Used by dlsym
     loadedLibsByHandle: {},
-    init: () => newDSO('__main__', {{{ cDefs.RTLD_DEFAULT }}}, wasmImports),
+    init: () => {
+#if ASSERTIONS
+      // This function needs to run after the intial wasmImports object
+      // as been created.
+      assert(wasmImports);
+#endif
+      newDSO('__main__', {{{ cDefs.RTLD_DEFAULT }}}, wasmImports);
```

Comment:
Do we not need a `return` here? There are `{ }` so I think we do?

Reply:
I'm not trying to return anything.. just setup the special DSO zero

Reply:
Oh, I see, so the return value can be ignored here. sgtm.